crates/cext/src/circuit.rs

```diff
@@ -561,3 +570,37 @@ pub unsafe extern "C" fn qk_opcounts_free(op_counts: OpCounts) {
         let _ = Box::from_raw(data);
     }
 }
+
+/// @ingroup QkCircuit
+/// Convert to a Python-space ``QuantumCircuit``
+///
+/// @param circuit the C-space ``QkCircuit`` pointer
+///
+/// @return A Python ``QuantumCircuit`` object
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to
+/// a ``QkCircuit``
+///
+/// It is assumed that the thread currently executing this function holds the
+/// Python GIL. This is required to create the Python object returned by this
+/// function.
+///
+/// This function takes ownership of the pointer and gives it to Python. Using
+/// the input ``circuit`` pointer after it's passed to this function is
+/// undefined behavior.
```

Comment:
It would be nice to move this warning to the main text, I went straight ahead and stepped into this problem lol (Also one could argue the `Safety` section is what you need to _call_ the function safely, not what happens later 😄)
```suggestion
/// Convert to a Python-space ``QuantumCircuit``.
///
/// This function takes ownership of the pointer and gives it to Python. Using
/// the input ``circuit`` pointer after it's passed to this function is
/// undefined behavior. In particular, ``qk_circuit_free`` should not be called
/// on this pointer anymore.
///
/// @param circuit The C-space ``QkCircuit`` pointer.
///
/// @return A Python ``QuantumCircuit`` object.
///
/// # Safety
///
/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to
/// a ``QkCircuit``
///
/// It is assumed that the thread currently executing this function holds the
/// Python GIL. This is required to create the Python object returned by this
/// function.
```